addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -94,27 +108,54 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
 
         long startTime = System.currentTimeMillis();
+        String site = getSite(msg);
+        tracker.putIfAbsent(site, new HashSet<String>());
+
         for (Application app : this.getApps()) {
-            ApplicationMatch appMatch = checkAppMatches(null, app, msg, source);
-            if (appMatch != null) {
-                String site = ExtensionWappalyzer.normalizeSite(msg.getRequestHeader().getURI());
-                LOGGER.debug("Adding {} to {}", app.getName(), site);
-                addApplicationsToSite(site, appMatch);
+            // Track matched based on site (authority)
+            synchronized (tracker) {
```

Comment:
Would also be good to have tests that ensure that we are not scanning again.

Reply:
Call scan on the same message/URL twice and assert number of alerts or techs?

Reply:
Yes.